gen-windows-sys-binding/src/main.rs

```diff
@@ -0,0 +1,52 @@
+//! Adapted from
+//! https://github.com/rust-lang/rust/blob/master/src/tools/generate-windows-sys/src/main.rs
+
+use std::{
+    fs,
+    io::{self, Write},
+};
+
+/// This is printed to the file before the rest of the contents.
+const PRELUDE: &str = r#"// This file is autogenerated.
+//
+// To add bindings, edit windows_sys.lst then run:
+//
+// ```
+// cd generate-windows-sys/
+// cargo run
+// ```
+"#;
+
+const POSTLUDE: &str = r#"
+/// Adapted from https://doc.rust-lang.org/src/core/ptr/mod.rs.html#600-607
+const fn invalid_mut<T>(addr: usize) -> *mut T {
+    addr as *mut T
```

Comment:
A comment that this would be `transmute()` except for MSRV requirement might be good.

Reply:
Thanks!
I've added comment about MSRV and the strict_provenance.